src/module/io.c

```diff
@@ -77,8 +88,13 @@ static bool handleRequestError(uv_fs_t* request)
 // Allocates a new request that resumes [fiber] when it completes.
 uv_fs_t* createRequest(WrenValue* fiber)
 {
-  uv_fs_t* request = (uv_fs_t*)malloc(sizeof(uv_fs_t));
-  request->data = fiber;
+  char* buffer = (char*)malloc(sizeof(uv_fs_t) + sizeof(FileReqData));
```

Comment:
Instead of a single concatenated alloc, I think it's clearer to allocate the `FileReqData` separately (and free it below).